src/placeos-driver/driver_model.cr

```diff
@@ -25,7 +25,8 @@ struct PlaceOS::Driver::DriverModel
       @functions = {} of String => Hash(String, Array(JSON::Any)),
       @implements = [] of String,
       @requirements = {} of String => Array(String),
-      @security = {} of String => Array(String)
+      @security = {} of String => Array(String),
+      @settings = {type: "object", properties: {} of String => JSON::Any, required: [] of String}
```

Comment:
just a small note, `required` cannot be an empty array, from the specs, it `MUST have at least one element`: [specs](https://tools.ietf.org/html/draft-wright-json-schema-validation-00#section-5.15).

Reply:
so basically when required is an empty array, or when it is nil, then we. need to ignore the field (don't serialise it)